From 64fa83c377c0b2e06c797f64f5816556461a33fa Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Tue, 22 Oct 2019 16:29:24 +0530
Subject: [PATCH 0314/1921] octeontx2-pf: Use napi_complete_done instead of
 napi_complete

Use napi_complete_done to allow for the use of gro_flush_timeout.

Change-Id: Ic9456cbcd147c5613d5aa99301a80084ab23dd57
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/17957
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
index adce012f1e51..3290696300d9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
@@ -497,10 +497,7 @@ int otx2_napi_handler(struct napi_struct *napi, int budget)
 	/* Clear the IRQ */
 	otx2_write64(pfvf, NIX_LF_CINTX_INT(cq_poll->cint_idx), BIT_ULL(0));
 
-	if (workdone < budget) {
-		/* Exit polling */
-		napi_complete(napi);
-
+	if (workdone < budget && napi_complete_done(napi, workdone)) {
 		/* If interface is going down, don't re-enable IRQ */
 		if (pfvf->intf_down)
 			return workdone;
-- 
2.31.1

